src/west/commands/project.py

```diff
@@ -212,38 +212,142 @@ def __init__(self):
             _wrap('''
             Updates all projects according to the manifest file, `west.yml`,
             in the manifest repository.
-            All projects will be checked out in detached HEAD and the local
-            manifest-rev branch will be reset hard to same HEAD.
 
-            Any local branches will not be altered by this command.
+            By default:
 
-            This command will not touch the manifest repository.
+            1. West itself is updated to the newest revision in the manifest
+            file.
+
+            2. The local per-project manifest-rev branches tracked by
+            west will be hard reset to the updated commits fetched
+            from each project's remote.
+
+            3. All projects will have detached HEADs checked out at
+            the new manifest-rev commits.
+
+            4. Any locally checked out project branches will be left
+            behind, unaltered. When this happens, this command
```

Comment:
I know `git` talks about leaving commits behind, when they are not on a branch, cause they will be garbage collected at some point, and thus being lost.

But I haven't heard anyone talking about, leaving a branch behind.
The branch will stay there, until deleted.

Reply:
How about combinining 3. and 4. into:

```
3. All projects will have detached HEADs checked out at
the new manifest-rev commits. Locally checked out branches
are unaffected but will no longer checked out by default.
West will print information on how to get back to those branches
or update them to the new manifest-rev versions.

You can avoid 1. using --no-update. You can (sometimes)
avoid 3. using --keep-descendants and/or --rebase.
See below for more information.
```

Reply:
yep.
Minor comments:
> `at the new manifest-rev commits`

suggestion, remove `new`

> but will no longer checked out by default.

should be `but will no longer be checked out by default.`

Reply:
Done, thanks

Reply:
:+1: